chainerio/filesystems/posix.py

```diff
@@ -24,9 +24,24 @@ def open(self, file_path, mode='r',
                        buffering, encoding, errors,
                        newline, closefd, opener)
 
-    def list(self, path_or_prefix: str = None):
-        for file in os.scandir(path_or_prefix):
-            yield file.name
+    def list(self, path_or_prefix: str = None, recursive=False):
+        if recursive:
+            path_or_prefix.rstrip("/")
+            # use len instead of len + 1 as root in os.walk does not end
+            # with "/"
+            prefix_end_index = len(path_or_prefix)
```

Comment:
Needs +1 here.

Reply:
My mistake, sorry. The above line should be `path_or_prefix = path_or_prefix.rstrip("/")`